torch/_refs/__init__.py

```diff
@@ -10,10 +10,11 @@
 from typing import Callable, List, Optional, overload, Sequence, Tuple, Union
 
 import torch
-
 import torch._prims as prims
 import torch._prims_common as utils
 from torch import sym_float, sym_int
+
+from torch._decomp.decompositions import pw_cast_for_int_to_real
```

Comment:
please don't use this wrappers in refs, there's `@elementwise_type_promotion_wrapper`

Reply:
```py
@elementwise_type_promotion_wrapper(
    type_promoting_args=("self"),
    type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.INT_TO_FLOAT,
)
```

This doesn't seem to work for this func. IIUC, it's because `elementwise_type_promotion_wrapper` converts to computational dtype while this op just casts to default FP type. From `logsumexp_out`:

```cpp
    if (at::isIntegralType(self.scalar_type(), /*includeBool=*/true)) {
      // for integral inputs, promote input to default floating type.
      auto default_dtype = at::typeMetaToScalarType(c10::get_default_dtype());
      logsumexp_out_impl(result, self.to(default_dtype), dims, keepdim);
```

Reply:
@elementwise_type_promotion_wrapper with correct parameters works for all refs including those converting int inputs to default FP type (like `log`, `exp` etc), I find it hard to believe that it doesn't work for this op.

Reply:
Well, I did try the above wrapper before posting. With it, the error that I'm seeing is ` RuntimeError: "abs_cpu" not implemented for 'Bool'`, which doesn't appear with `pw_cast_for_int_to_real`.

Reply:
Ah, nevermind. It's the comma after "self", which matters.